src/data/media-player.ts

```diff
@@ -317,5 +319,19 @@ export const computeMediaControls = (
     });
   }
 
+  if (state === "on" && supportsFeature(stateObj, SUPPORT_NEXT_CHANNEL)) {
+    buttons.push({
+      icon: "mdi:chevron-up",
+      action: "media_next_channel",
+    });
+  }
+
+  if (state === "on" && supportsFeature(stateObj, SUPPORT_PREVIOUS_CHANNEL)) {
+    buttons.push({
+      icon: "mdi:chevron-down",
+      action: "media_previous_channel",
+    });
+  }
+
   return buttons.length > 0 ? buttons : undefined;
```

Comment:
I think it should check if there is a channel active before showing buttons. A media player could have support for channels, but not actively at a channel list.

Reply:
This is implicit checked, since a TV, when it is in classical TV mode it is in state "on", but when playing some media or streaming app it is in state "playing" and those channel switchers are not shown

Reply:
The media player should update its `supported_features` attribute according to what is currently supported, we should not condition it by checking the state.

Reply:
supported_features is not expected to change at runtime. It should be static once added to hass has executed.

Reply:
I don't think we have such a limitation documented anywhere, and the cast integration for example changes supported features depending on the app currently running (some apps support media controls, some don't).

Reply:
Supported features are used to configure Alexa and Google Assistant. They are not re-configured if they happen to change.

Reply:
That's a valid point, but we should then either document that supported features are not allowed to change or teach Alexa and Google Assistant to handle updated supported features.

Reply:
Supported features are not supposed to change during runtime, but media players are "grandfathered" in as they had the feature first. For media player it would make more sense if we split "media player" features and the "current media" capabilities.

Google and Alexa are indeed not re-configured, nor do we update the entity registry (that caches the supported features in case the entity is not provided after a restart)